### PR TITLE
docs: add description about CI platforms to CONTRIBUTE.md

### PR DESCRIPTION
**Pull request by mback2k**
An attempt to clarify the current CI platform state in the docs.

### CODE REVIEW

**Review by bagder — APPROVED** [review]
Excellent stuff!

**Review by MarcelRaad — APPROVED** [review]
👍

File: docs/CONTRIBUTE.md [hunk 1]
```diff
@@ -172,6 +172,33 @@ you are expected to fix the problem. If you don't understand when the issue is
...
+ | LGTM analysis:  C/C++               | stable | new findings               |
+ | buildbot/curl_winssl_ ...           | stable | all errors and failures    |
+ | continuous-integration/appveyor/pr  | stable | all errors and failures    |
+ | continuous-integration/travis-ci/pr | stable | all errors and failures    |
+ | curl.curl (linux ...)               | stable | all errors and failures    |
+ | curl.curl (windows ...)             | flaky  | common errors and failures |
+ | deepcode-ci-bot                     | stable | new findings               |
+ | musedev                             | stable | new findings               |
+
+Sometimes the tests fail due to a dependency service temporarely being offline
```
Comment (jay) [inline docs/CONTRIBUTE.md:194]:
```suggestion
Sometimes the tests fail due to a dependency service temporarily being offline
```

File: docs/CONTRIBUTE.md [hunk 2]
```diff
@@ -172,6 +172,33 @@ you are expected to fix the problem. If you don't understand when the issue is
...
+or otherwise unavailable, eg. package donwloads. In this case you can just
```
Comment (jay) [inline docs/CONTRIBUTE.md:195]:
```suggestion
or otherwise unavailable, eg. package downloads. In this case you can just
```

**Comment by mback2k** [comment]
> I'd just say it as windows testing is flaky at the moment rather than a whole chart.

I prefer the whole chart to give detailed guidance for newcomers.

> Also "common errors and failures" is not defined (is it?). Anyway, I'm fine with it once the typos are addressed.

I don't really mean "common" in the sense of "general" here, I mean it more like "recurring over multiple CI runs and variants". I tried to update the short description a little to hopefully make this clearer. Please take a look.

**Pull request closed without merging**